<launch>
    <arg name="sim" default="false"/>

    <arg name="gui" default="true"/>
    <arg name="update_description" default="true"/>
    <arg name="verbose" default="false"/>
    <arg name="world_name" default="$(find arm_robots)/world/empty_with_moveit_plugin.world"/>

    <group ns="static_transform_publishers">
        <node pkg="tf2_ros" type="static_transform_publisher" name="rename_origin" required="true"
              args="0 0 0 0 0 0 robot_root world">
        </node>
    </group>

    <group if="$(arg update_description)">
        <!-- uploads robot description and setups up other parameters -->
        <include file="$(find thanos_moveit_config)/launch/planning_context.launch">
            <arg name="load_robot_description" value="true"/>
        </include>
    </group>

    <group unless="$(arg sim)">
        <!-- Start gazebo -->
        <!-- <include file="$(find gazebo_ros)/launch/empty_world.launch"> -->
        <!--     <arg name="world_name" value="$(arg world_name)"/> -->
        <!--     <arg name="gui" value="$(arg gui)"/> -->
        <!--     <arg name="paused" value="false"/> -->
        <!--     <arg name="verbose" value="$(arg verbose)"/> -->
        <!-- </include> -->

        <!-- to let moveit send trajectories with the real robot -->
        <include file="$(find arm_robots)/launch/trajectory_follower.launch">
            <arg name="use_thanos" value="true"/>
        </include>

        <!-- ROS control for real victor, does nothing at the moment -->
        <!-- <include file="$(find victor_control)/launch/victor_control.launch"> -->
        <!--     <arg name="robot_namespace" value="victor"/> -->
        <!-- </include> -->

        <!-- Move it-->
        <include ns="victor" file="$(find thanos_moveit_config)/launch/move_group.launch">
            <arg name="allow_trajectory_execution" value="true"/>
            <arg name="moveit_controller_manager" value="real"/>
        </include>
    </group>
</launch>
